packages/app/src/sections/futures/TraderHistory.tsx

```diff
@@ -73,7 +75,23 @@ const TraderHistory: FC<TraderHistoryProps> = memo(
 						  i.status.toLowerCase().includes(searchTerm)
 						: true
 				)
-		}, [positionHistory, positions, searchTerm])
+
+			// export data to CSV
+			let csvData =
+				'Date/Time,Market,Status,Trades,Total Volume,Realized P&L USD,Realized P&L %,Funding,Tx Hash\n'
+			result.forEach(
+				(row: (typeof result)[number]) =>
+					(csvData += `${new Date(row.timestamp).toISOString()},${row.marketShortName},${
+						row.status
+					},${row.trades},${row.totalVolume.toNumber()},${row.pnl.toNumber()},${row.pnlPct.slice(
+						1,
+						-2
+					)},${row.funding.toNumber()},${row.transactionHash}\n`)
+			)
+			dispatch(setCsvExportData(csvData))
```

Comment:
It would be nice if we could move all of this data formatting logic out of the component and into redux. So I think we could create a selector for the table data `selectPositionHistoryTableData` which we move the original table data formatting to and then `selectCsvExport` can just pull in the data from `selectPositionHistoryTableData` meaning we can also remove the `csvExport` field it's reducer.